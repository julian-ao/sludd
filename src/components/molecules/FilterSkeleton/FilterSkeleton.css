.filter_container {
    color: #6d6d6d;
    background-color: white;
    border-radius: 10px;
    padding: 1rem;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: 0.2s;
    cursor: pointer;
    height: min-content;
}

.filter_container:hover {
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
}

.filter_title {
    display: flex;
    align-items: center;
    cursor: pointer;
}

.filter_values_container {
    visibility: hidden;
    height: 0;
}

.filter_value {
    display: flex;
    align-items: center;
    padding: 5px 0;
}

.filter_button {
    margin-right: 10px;
    margin-left: 0;
    height: 20px;
    width: 20px;
}

.filter_icon {
    margin-right: 10px;
    width: 20px;
}
