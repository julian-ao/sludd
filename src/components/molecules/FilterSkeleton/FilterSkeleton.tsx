import { useState } from 'react';
import './filterSkeleton.css';

export type FilterSkeletonProps = {
    title: string;
    values: string[];
    currentFilters: string[]; // state from parent component
    setFilters: (list: string[]) => void; // state setter from parent component
};

const FilterSkeleton = (props: FilterSkeletonProps) => {

    const [visible, setVisible] = useState<boolean>(false);

    const handleCheckboxClick = (event: React.MouseEvent<HTMLInputElement>) => {
        event.stopPropagation(); // Prevent the click event from propagating to the parent container

        const value = event.currentTarget.value;
        if (props.currentFilters.includes(value)) {
            props.setFilters(props.currentFilters.filter(item => item !== value)); // remove value
        } else {
            props.setFilters([...props.currentFilters, value]); // add value
        }
    };

    return (
        <div
            className='filter_container'
            onClick={() => setVisible(!visible)}
        >
            <div className='filter_title' style={{ borderBottom: visible ? '1px solid #c7c7c7' : 'none', paddingBottom: visible ? '10px' : 0 }}>
                <svg xmlns="http://www.w3.org/2000/svg" className='filter_icon' viewBox="0 0 20 18" fill="none">
                    <path d="M0.152157 0.894636C0.41 0.347697 0.956939 0 1.56248 0H18.4395C19.045 0 19.5919 0.347697 19.8498 0.894636C20.1076 1.44158 20.0295 2.08618 19.6466 2.55499L12.5013 11.2865V16.2519C12.5013 16.7246 12.2356 17.1583 11.8098 17.3692C11.3839 17.5802 10.88 17.5372 10.501 17.252L8.00073 15.3768C7.68429 15.1424 7.50067 14.7713 7.50067 14.3767V11.2865L0.3514 2.55108C-0.0275511 2.08618 -0.109592 1.43767 0.152157 0.894636Z" fill="#6D6D6D" />
                </svg>
                {props.title}
                {props.currentFilters.length > 0 && (
                    <div style={{ marginLeft: '5px' }}>({props.currentFilters.length} valgt)</div>
                )}
            </div>
            <div className='filter_values_container' style={{ visibility: visible ? 'visible' : 'hidden', height: visible ? 'auto' : 0, paddingTop: visible ? '10px' : 0 }}>
                {props.values.map((value, index) => {
                    return (
                        <label key={index} className='filter_value' onClick={(event) => { event.stopPropagation() }}>
                            <input value={value} type={'checkbox'} className='filter_button' onClick={handleCheckboxClick} />
                            {value}
                        </label>
                    );
                })}
            </div>
        </div>
    );
};

export default FilterSkeleton;
