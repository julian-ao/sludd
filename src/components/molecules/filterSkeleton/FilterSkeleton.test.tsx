import { cleanup, render } from '@testing-library/react'
import { afterEach } from 'vitest'
import FilterSkeleton from './FilterSkeleton'

afterEach(() => {
    cleanup()
})
describe('FilterSkeleton', () => {
    it('should render correctly', () => {
        const { container } = render(<FilterSkeleton
            title={'Filter'}
            values={['Adressenavn', 'By', 'Bydel', 'Gard', 'Forretningsbygg', 'Bruk', 'Tjern', 'Skole']}
            currentFilters={['Adressenavn', 'By', 'Bydel', 'Gard', 'Forretningsbygg', 'Bruk', 'Tjern', 'Skole']}
            setFilters={() => { }}
        />)
        expect(container).toMatchSnapshot()
    }),
        it('should render correctly with no current filters', () => {
            const { container } = render(<FilterSkeleton
                title={'Filter'}
                values={['Adressenavn', 'By', 'Bydel', 'Gard', 'Forretningsbygg', 'Bruk', 'Tjern', 'Skole']}
                currentFilters={[]}
                setFilters={() => { }}
            />)
            expect(container).toMatchSnapshot()
        }
        )
})