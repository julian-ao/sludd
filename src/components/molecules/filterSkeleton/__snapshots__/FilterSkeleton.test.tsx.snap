// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterSkeleton > should render correctly 1`] = `
<div>
  <div
    class="filter_container"
  >
    <div
      aria-label="click me!"
      class="filter_box"
      role="filter_box"
    >
      <div
        class="filter_title"
        style="padding-bottom: 0px;"
      >
        <svg
          class="filter_icon"
          fill="none"
          viewBox="0 0 20 18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0.152157 0.894636C0.41 0.347697 0.956939 0 1.56248 0H18.4395C19.045 0 19.5919 0.347697 19.8498 0.894636C20.1076 1.44158 20.0295 2.08618 19.6466 2.55499L12.5013 11.2865V16.2519C12.5013 16.7246 12.2356 17.1583 11.8098 17.3692C11.3839 17.5802 10.88 17.5372 10.501 17.252L8.00073 15.3768C7.68429 15.1424 7.50067 14.7713 7.50067 14.3767V11.2865L0.3514 2.55108C-0.0275511 2.08618 -0.109592 1.43767 0.152157 0.894636Z"
            fill="#6D6D6D"
          />
        </svg>
        Filter
        <div
          style="margin-left: 5px;"
        >
          (
          8
           valgt)
        </div>
      </div>
      <div
        class="filter_values_container"
        role="filter_values_container"
        style="visibility: hidden; height: 0px; padding-top: 0px;"
      >
        <label
          class="filter_value"
        >
          <input
            checked=""
            class="filter_button"
            type="checkbox"
            value="Adressenavn"
          />
          Adressenavn
        </label>
        <label
          class="filter_value"
        >
          <input
            checked=""
            class="filter_button"
            type="checkbox"
            value="By"
          />
          By
        </label>
        <label
          class="filter_value"
        >
          <input
            checked=""
            class="filter_button"
            type="checkbox"
            value="Bydel"
          />
          Bydel
        </label>
        <label
          class="filter_value"
        >
          <input
            checked=""
            class="filter_button"
            type="checkbox"
            value="Gard"
          />
          Gard
        </label>
        <label
          class="filter_value"
        >
          <input
            checked=""
            class="filter_button"
            type="checkbox"
            value="Forretningsbygg"
          />
          Forretningsbygg
        </label>
        <label
          class="filter_value"
        >
          <input
            checked=""
            class="filter_button"
            type="checkbox"
            value="Bruk"
          />
          Bruk
        </label>
        <label
          class="filter_value"
        >
          <input
            checked=""
            class="filter_button"
            type="checkbox"
            value="Tjern"
          />
          Tjern
        </label>
        <label
          class="filter_value"
        >
          <input
            checked=""
            class="filter_button"
            type="checkbox"
            value="Skole"
          />
          Skole
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`FilterSkeleton > should render correctly with no current filters 1`] = `
<div>
  <div
    class="filter_container"
  >
    <div
      aria-label="click me!"
      class="filter_box"
      role="filter_box"
    >
      <div
        class="filter_title"
        style="padding-bottom: 0px;"
      >
        <svg
          class="filter_icon"
          fill="none"
          viewBox="0 0 20 18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0.152157 0.894636C0.41 0.347697 0.956939 0 1.56248 0H18.4395C19.045 0 19.5919 0.347697 19.8498 0.894636C20.1076 1.44158 20.0295 2.08618 19.6466 2.55499L12.5013 11.2865V16.2519C12.5013 16.7246 12.2356 17.1583 11.8098 17.3692C11.3839 17.5802 10.88 17.5372 10.501 17.252L8.00073 15.3768C7.68429 15.1424 7.50067 14.7713 7.50067 14.3767V11.2865L0.3514 2.55108C-0.0275511 2.08618 -0.109592 1.43767 0.152157 0.894636Z"
            fill="#6D6D6D"
          />
        </svg>
        Filter
      </div>
      <div
        class="filter_values_container"
        role="filter_values_container"
        style="visibility: hidden; height: 0px; padding-top: 0px;"
      >
        <label
          class="filter_value"
        >
          <input
            class="filter_button"
            type="checkbox"
            value="Adressenavn"
          />
          Adressenavn
        </label>
        <label
          class="filter_value"
        >
          <input
            class="filter_button"
            type="checkbox"
            value="By"
          />
          By
        </label>
        <label
          class="filter_value"
        >
          <input
            class="filter_button"
            type="checkbox"
            value="Bydel"
          />
          Bydel
        </label>
        <label
          class="filter_value"
        >
          <input
            class="filter_button"
            type="checkbox"
            value="Gard"
          />
          Gard
        </label>
        <label
          class="filter_value"
        >
          <input
            class="filter_button"
            type="checkbox"
            value="Forretningsbygg"
          />
          Forretningsbygg
        </label>
        <label
          class="filter_value"
        >
          <input
            class="filter_button"
            type="checkbox"
            value="Bruk"
          />
          Bruk
        </label>
        <label
          class="filter_value"
        >
          <input
            class="filter_button"
            type="checkbox"
            value="Tjern"
          />
          Tjern
        </label>
        <label
          class="filter_value"
        >
          <input
            class="filter_button"
            type="checkbox"
            value="Skole"
          />
          Skole
        </label>
      </div>
    </div>
  </div>
</div>
`;
